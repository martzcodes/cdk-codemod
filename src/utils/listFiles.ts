import { readdirSync } from "fs";

// recursively list all files in a directory and its subdirectories
export const listFiles = (
  dir: string,
  fileList: string[] = [],
  extension: string
) => {
  const files = readdirSync(dir, { withFileTypes: true });
  files.forEach((file) => {
    if (file.name === "node_modules") return;
    if (file.isDirectory()) {
      fileList = listFiles(`${dir}/${file.name}`, fileList, extension);
    } else {
      if (file.name.endsWith(extension)) {
        fileList.push(`${dir}/${file.name}`);
      }
    }
  });
  return fileList;
};
