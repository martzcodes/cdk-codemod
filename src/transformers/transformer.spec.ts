import { readFile } from "fs/promises";
import path from "path";
import jscodeshift from "jscodeshift";
import { describe, expect, it } from "vitest";
import { listFiles } from "../utils/listFiles";

import transformer from "./transformer";

describe("transformer", () => {
  const paths = listFiles(path.resolve("src/transformers/__fixtures__"), [], ".ts");
  it.each(paths)(
    "%s",
    async (inputPath) => {
      const inputCode = await readFile(inputPath, "utf8");
      const input = { path: inputPath, source: inputCode };

      const output = await transformer(input, {
        j: jscodeshift,
        jscodeshift,
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        stats: () => {},
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        report: () => {},
      });
      expect(output.trim()).toMatchSnapshot();
    },
    100000
  );
});
