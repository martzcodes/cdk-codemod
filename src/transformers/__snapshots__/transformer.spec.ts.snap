// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformer > /Users/mattmartz/dev/cdk-codemod/src/transformers/__fixtures__/example.ts 1`] = `
"import { App, Stack, StackProps } from \\"aws-cdk-lib\\";
import { Construct } from \\"constructs\\";

export class SomeConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
  }
}

const app = App();
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    new SomeConstruct(this, \\"SomeConstruct\\");
  }
}

new MyStack(app, \\"MyStack\\");"
`;

exports[`transformer > /Users/mattmartz/dev/cdk-codemod/src/transformers/__fixtures__/example-named.ts 1`] = `
"/* eslint-disable import/no-unresolved */
import { App, Stack, StackProps } from \\"aws-cdk-lib\\";

import { Construct } from \\"constructs\\";

export class SomeConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
  }
}

const app = App();
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    new SomeConstruct(this, \\"SomeConstruct\\");
  }
}

new MyStack(app, \\"MyStack\\");"
`;

exports[`transformer > /Users/mattmartz/dev/cdk-codemod/src/transformers/__fixtures__/example-with-bucket.ts 1`] = `
"/* eslint-disable import/no-unresolved */
import { Bucket } from \\"aws-cdk-lib/aws-s3\\";
import { App, Stack, StackProps } from \\"aws-cdk-lib\\";
import { Construct } from \\"constructs\\";

export class SomeConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    new Bucket(this, \\"MyBucket\\");
  }
}

const app = App();
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    new SomeConstruct(this, \\"SomeConstruct\\");
  }
}

new MyStack(app, \\"MyStack\\");"
`;
